src/DepositHandler.sol

```diff
@@ -50,7 +83,7 @@ contract DepositHandler is Pausable, AccessControl, IDepositHandlerErrors {
     {
         depositAmount = _depositAmount;
         depositToken = IERC20(_depositToken);
-        bootcampStartTime = block.timestamp;
+        bootcampStartTime = block.timestamp + DEPOSIT_STAGE_DURATION; // set 2 days from the bootcamp deployment for depositing.
```

Comment:
Sorry if I missed the conversations about it, please feel free to reference/ copy paste / tell me it was on discord... any discussions. 

I think this is only really useful in case of whitelisting deposit, but we can assume that we don't need a deposit stage duration. When the bootcamp contract is deployed, we can. start to accept deposits. When it starts, we cannot anymore. 

Indeed encode is giving you a time window... but if we don't have on-chain whitelisting, then this time window have to also be off-chain. I strongly understand the point of view, but I think we can indeed keep it simple there and maybe not need this constant ?

Reply:
Yes we removed a whitelisting mechanism, but this is wrong - `When the bootcamp contract is deployed, we can. start to accept deposits. When it starts, we cannot anymore.`. It is wrong, because when a bootcamp instance is deployed then the time for bootcamp starts countdown already. Check the constructor and see if we'll remove a `+ DEPOSIT_STAGE_DURATION;` - then this will mean that the bootcamp is already in progress, but users will only start depositing(because the instance was created like couple of minutes ago).
Someone can deposit not on the first day, but on the second for example, and in this way user will skip 1 day of the bootcamp?

I don't see any complexity here. In this way nobody will join the bootcamp in the middle of it. Without this restriction user who was allowed for a deposit will be able to do it all the time the bootcamp going, and in the end it will be quite difficult to solve such cases

Reply:
Yes indeed this is wrong from your code perspective but not from a business logic perspective (like the general functionnalities perspectives). I think we shouldn't leave a two days specific time. I thought this was related to the 48 days time to deposit. But encode is allowing people at different time - not everyone at the same time. So for me the DEPOSIT_STAGE_DURATION needs to be calculated based on the date of the start of the bootcamp and the current block.timestamp at deployment.

Reply:
The 48h limits can be handled off-chain by encode (so in the same way as the whitelisting on the app). And without whitelisting, cannot be handle on-chain unfortunately